lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -2184,3 +2171,27 @@ void JavascriptObject::SpreadObjectLiteral(Var source, Var to, ScriptContext* sc
     AssertOrFailFast(succeeded);
     CopyDataPropertiesHelper<false, false>(source, target, scriptContext);
 }
+
+void JavascriptObject::Restify(Var source, Var to, Var excludedStatic, Var excludedComputed, ScriptContext* scriptContext)
```

Comment:
I know `Var` is just a typedef to `void*`, but it has some extra semantic meaning: a `Var` should be something that user code in Javascript could interact with (a tagged int, a floating point number, or a pointer to some JS data type). PropertyIdArray is an internal engine data type, so I'd prefer to write this as `void*` (assuming that we can't write it as `PropertyIdArray*` which would of course be the most descriptive option).

Reply:
Same advice for other things that deal with `PropertyIdArray` like `OP_StPropIdArrFromVar`

---
In reply to: [204247038](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r204247038) [](ancestors = 204247038)

Reply:
Maybe I've misinterpreted the general guidance here, since I just realized that `GetNonVarReg` returns `Var`. I feel like this is still a good idea, but I'm no longer confident that I've represented a pre-existing convention.

---
In reply to: [204247124](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r204247124) [](ancestors = 204247124,204247038)

Reply:
Hmmm but wouldn't it get confusing since `GetNonVarReg` and `GetReg` both return `Var` and directly pass them into the helper? I do agree though, Var seems like it should just be script-related data, but then `GetNonVarReg` should return `void*`.

---
In reply to: [204247236](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r204247236) [](ancestors = 204247236,204247124,204247038)

Reply:
It's true that we try generally to have Var be either a tagged value or a RecyclableObject*; there's a few cases where this doesn't hold, but not as many as you might think.

Reply:
`GetReg` does some extra validation (on debug builds) that `GetNonVarReg` doesn't - presumably because it expects the caller to know what they are doing. 

As far as this method is concerned, I would agree with Seth that those two property id lists should be of type `PropertyIdArray*`

---
In reply to: [204619121](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r204619121) [](ancestors = 204619121,204247236,204247124,204247038)